src/Components/Views/EditUser.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react'
+import { UserForm } from '../shared/Form'
+import { CreateUser, UpdateUser } from '../../validations/basic/user.dto'
+import { createValidator } from '../../utils/class-validator-formik'
+import { UserService } from '../../services/basics/user.service'
+import { useMutation } from 'react-query'
+import { User } from '@/types/types'
+import { useHistory } from 'react-router-dom'
+
+interface EditUserProps {
+  onClose: () => void;
+  location: {
+    state: {
+      initialValues: User;
+    };
+  };
+}
+
+export const EditUser: React.FC<EditUserProps> = ({ location, onClose }) => {
+  console.log('EditUser - initialValues:', location.state.initialValues)
+  console.log('EditUser - onClose:', onClose)
+  const validate = createValidator(location.state.initialValues instanceof CreateUser ? CreateUser : UpdateUser)
```

Comment:
its a lot of code for an argument, extract it into a variable and pass it to the function.

Reply:
Also I don't get why we have an `AddUser.tsx` File and here we also have the logic to add a user, I think you might want to delete unused code here.